File: data/data.code.javascript.client/ReqSceneEnterFly.js
module.exports = ReqSceneEnterFly;

var Packet = require('../net/Packet');


function ReqSceneEnterFly() {
	this._map_id = 0;


	this.Encode = function() {
		var packet = new Packet();
		packet.WriteUint(this._map_id);
		packet.Encode(2010);
		return packet;
	}
}

ReqSceneEnterFly.prototype = {
	set MapId(val) {
		this._map_id = val;
	},
	get MapId() {
		return this._map_id;
	},
}

File: examples/lua/client/protocols/ReqSceneEnterFly.lua
module("ReqSceneEnterFly", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnterFly})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.map_id = 0
end


function encode(self)
	 local pack = packet.new()
	pack:writeU32(self.map_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER_FLY)
end
